packages/sanity/src/_internal/cli/actions/deploy/undeployAction.ts

```diff
@@ -1,23 +1,32 @@
 import {type CliCommandArguments, type CliCommandContext} from '@sanity/cli'
 
-export default async function undeployStudio(
-  args: CliCommandArguments<Record<string, unknown>>,
+import {debug as debugIt} from '../../debug'
+import {deleteUserApplication, getUserApplication} from './helpers'
+
+const debug = debugIt.extend('undeploy')
+
+export default async function undeployStudioAction(
+  _: CliCommandArguments<Record<string, unknown>>,
   context: CliCommandContext,
 ): Promise<void> {
-  const {apiClient, chalk, output, prompt} = context
+  const {apiClient, chalk, output, prompt, cliConfig} = context
 
   const client = apiClient({
     requireUser: true,
     requireProject: true,
-  })
+  }).withConfig({apiVersion: 'v2024-08-01'})
 
   // Check that the project has a studio hostname
   let spinner = output.spinner('Checking project info').start()
-  const project = await client.projects.getById(client.config().projectId as string)
-  const studioHost = project && project.studioHost
+
+  const userApplication = await getUserApplication({
+    client,
+    appHost: cliConfig && 'studioHost' in cliConfig ? cliConfig.studioHost : undefined,
+  })
+
   spinner.succeed()
 
-  if (!studioHost) {
+  if (!userApplication) {
     output.print('Your project has not been assigned a studio hostname.')
```

Comment:
I found this output a little bit confusing -- it could also just mean that they DO have the hostname but DON'T have a default and didn't put it in CLI (which will happen to every one starting a new project from here on out.)

```suggestion
    output.print('Your project has not been assigned a studio hostname,')
     output.print('or you do not have studioHost set in in sanity.cli.js or sanity.cli.ts.')
```